src/cmd/list.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright © 2019 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/containers/toolbox/pkg/podman"
+	"github.com/containers/toolbox/pkg/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var listFlags struct {
+	listContainers bool
+	listImages     bool
+}
+
+var listCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List existing toolbox containers and images",
+	Run: func(cmd *cobra.Command, args []string) {
+		list(cmd, args)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(listCmd)
+
+	flags := listCmd.Flags()
+	flags.BoolVarP(&listFlags.listContainers, "containers", "c", false, "List only toolbox containers, not images")
+	flags.BoolVarP(&listFlags.listImages, "images", "i", false, "List only toolbox images, not containers")
+}
+
+func list(cmd *cobra.Command, args []string) error {
+	if !listFlags.listContainers && !listFlags.listImages {
+		listFlags.listContainers = true
+		listFlags.listImages = true
+	}
+
+	var images []map[string]interface{}
+	var containers []map[string]interface{}
+	var err error
+
+	if listFlags.listImages {
+		images, err = GetImages()
+		if err != nil {
+			logrus.Error(err)
+			logrus.Debugf("%+v", err)
+		}
+	}
+
+	if listFlags.listContainers {
+		containers, err = GetContainers()
+		if err != nil {
+			logrus.Error(err)
+			logrus.Debugf("%+v", err)
+		}
+	}
+
+	err = outputList(images, containers)
+	if err != nil {
+		logrus.Fatal(err)
+		logrus.Fatalf("%+v", err)
+	}
+
+	return err
+}
+
+func GetContainers() ([]map[string]interface{}, error) {
+	logrus.Info("Fetching containers with label=com.github.debarshiray.toolbox=true")
+	args := []string{"-a", "--filter", "label=com.github.debarshiray.toolbox=true"}
```

Comment:
Typo alert! Should be `label=com.redhat.component=fedora-toolbox"`

Reply:
True. But shouldn't all three labels be used?